tests/schemas/builder.py

```diff
@@ -117,14 +135,113 @@ def generate_core_data_specs():
     return {data_cls.__name__: data_class_to_spec(data_cls) for data_cls in data_classes}
 
 
+def collect_cmd_serializers_from_module(mod):
+    data_classes = []
+    for item_name in dir(mod):
+        item = getattr(mod, item_name)
+        # Ignore non-data classes
+        if not isinstance(item, type) or not issubclass(item, _BASE_DATA_CLASSES):
+            continue
+        if item in _BASE_DATA_CLASSES:
+            continue
+        # Data classes with default serializer cannot be serialized, hence no need
+        # to check them (note they will be checked if they are used in Nested field)
+        if item.SERIALIZER_CLS is BaseSerializer:
+            continue
+        # Ignore imported classes (avoid to populate current module collection
+        # with extenal imported schema.
+        # Example: Avoid to add imported api schemas while generating parsec.core.types)
+        if not item.__module__.startswith(mod.__name__):
+            continue
+        data_classes.append(item)
+    return data_classes
+
+
+def cmd_serializer_to_spec(cmd_serializer: CmdSerializer):
+    assert isinstance(cmd_serializer, CmdSerializer)
+    # Serialization should always be msgpack, but add it just to be sure...
+    return {
+        "req": {
+            "serializing": _SERIALIZER_TO_STR[type(cmd_serializer._req_serializer)],
+            **schema_to_spec(cmd_serializer._req_serializer.schema),
+        },
+        "rep": {
+            "serializing": _SERIALIZER_TO_STR[type(cmd_serializer._rep_serializer)],
+            **schema_to_spec(cmd_serializer.rep_noerror_schema),
+        },
+    }
+
+
+def generate_api_protocol_specs():
+    # First collect all command serializers
+    from parsec.api import protocol as protocol_mod
+
+    cmd_serializers = {}
+    for item_name in dir(protocol_mod):
+        item = getattr(protocol_mod, item_name)
+        if not isinstance(item, CmdSerializer):
+            continue
+        # This is where things start to be hacky: given we don't use enum
+        # for command name, and given we sometime use the same schema for
+        # APIv1 and v2 definitions, we rely on a weak naming convention
+        # to do the identification...
+        match = re.match(r"^(?P<name>(apiv1_)?[a-z][a-z0-9_]+)_serializer$", item_name)
```

Comment:
So those regex define what is a valid name for different parts of the API? Maybe it would makes sense to define them together somewhere else with some comments and some examples of valid names to make it clearer.

Reply:
> So those regex define what is a valid name for different parts of the API?

yup, this is totally sub-optimal, but otherwise we would have to change plenty of code in the api.
The good think is we check naming for all `CmdSerializer`, so a wrong naming will be caught here (hence it should be obvious enough how to fix the naming) and given we also check unused `CmdSerializer` we should be also safe against name shadowing.

Note we are not defining a scheme for the api command, but for the variables implementing them. So I don't think it's not that bad to have this kind of hack ;-)